<launch>
  <arg name="scene_name" default="cofusion_car" />
  <arg name="sensor_name" default="cofusion_car" />
  <arg name="gt_segmentation" default="false" />
  <arg name="visualize" default="true" />

	<node pkg="tsdf_plusplus_ros" type="tsdf_plusplus_node" name="tsdf_plusplus_node" output="screen">
		<rosparam command="load" file="$(find tsdf_plusplus_ros)/config/default.yaml" />
	  <rosparam command="load" file="$(find tsdf_plusplus_ros)/config/$(arg scene_name).yaml" />
    <rosparam command="load" file="$(find rgbd_segmentation)/config/$(arg sensor_name).yaml" />

    <!-- Load the human-readable Microsoft COCO 80 object categories. -->
		<rosparam command="load" file="$(find tsdf_plusplus_ros)/config/coco_classes.yaml" />

    <!-- Settings for relying on ground truth 2D segmentation. -->
    <param name="segment_pointcloud_topic" value="/camera/object_segment" if="$(arg gt_segmentation)" />
    <param name="using_ground_truth_segmentation" value="true" if="$(arg gt_segmentation)" />

    <param name="visualizer/enable" value="$(arg visualize)" />
    <param name="meshing/update_mesh_every_n_sec" value="0.0" unless="$(arg visualize)" />

  </node>
</launch>
